src/sentry/integrations/gitlab/client.py

```diff
@@ -258,9 +258,15 @@ def check_file(self, repo, path, ref):
         Path requires file path and ref
         file_path must also be URL encoded Ex. lib%2Fclass%2Erb
         """
-        self.base_url = self.metadata["base_url"]
-        project_id = repo.config["project_id"]
-        encoded_path = quote(path, safe="")
-
-        request_path = GitLabApiClientPath.file.format(project=project_id, path=encoded_path)
-        return self.head_cached(request_path, params={"ref": ref})
+        try:
+            self.base_url = self.metadata["base_url"]
+            project_id = repo.config["project_id"]
+            encoded_path = quote(path, safe="")
+
+            request_path = GitLabApiClientPath.file.format(project=project_id, path=encoded_path)
+            return self.head_cached(request_path, params={"ref": ref})
+        except ApiError as e:
+            # Gitlab can return 404 or 400 if the file doesn't exist
+            if e.code not in [404, 400]:
```

Comment:
@NisanthanNanthakumar can we just leave the check for the `404` in the parent method?